index.js

```diff
@@ -4,68 +4,121 @@ var logUtil = require('./logUtil');
 var linesUtil = require('./linesUtil');
 var config = require('./config');
 
+function getLogLevel() {
+	var level = process.env.LOG_LEVEL;
+	if (level) {
+		level = config._logLevels[level.toLowerCase()];
+	} else {
+		//default log level
+		level = 7;
+	}
+	return level;
+}
+
+function meetsLogLevelRequirement(loggerLevel) {
+	return config._logLevels[loggerLevel] <= getLogLevel();
+}
+
 var m = {
 	//////////////////
 	// main methods //
 	//////////////////
 
 	// default
 	d: function() {
-		logUtil.logWithColor(config.colors.default, arguments);
+		if (meetsLogLevelRequirement('default')) {

```

Comment:
hopefully will move check to a function so not everywhere we have an if... might be something like:
```logIfMeetsLogLevel(functionLogLevel, loggingFunc, color, arguments)```

But, its up to you...just thinking out loud here xD